### PR TITLE
log the actual exceptions for metadata retry paths. 

### PR DESCRIPTION
**Pull request by adepue**
Catching blanket/non-specific exceptions is problematic, so at least catch an exception type and log what you saw for reference (in verbose mode).

## Description

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by adepue** [comment]
@narrieta for awareness/review.

File: azurelinuxagent/common/protocol/metadata.py [hunk 1]
```diff
@@ -314,8 +314,9 @@ def update_goal_state(self, forced=False, max_retry=3):
             try:
                 self.update_certs()
                 return
-            except:
+            except Exception as e:
                 logger.verbose("Incarnation is out of date. Update goalstate.")
+                logger.verbose("Exception: %{0}".format(e))
```
Comment (vrdmr) [inline azurelinuxagent/common/protocol/metadata.py]: Logging the traceback would help as well to identify the issue. Reference - https://github.com/Azure/WALinuxAgent/blob/master/azurelinuxagent/ga/exthandlers.py#L219.

Comment (adepue) [reply azurelinuxagent/common/protocol/metadata.py]: @vrdmr updated.  Let me know what you think.  I ended up copying the get_traceback function over... copy and paste not ideal, but I didn't want a crazy import loop either to get it.  I am happy to move it out to a utils method if you'd like?

File: azurelinuxagent/common/protocol/metadata.py [hunk 2]
```diff
@@ -314,8 +323,10 @@ def update_goal_state(self, forced=False, max_retry=3):
             try:
                 self.update_certs()
                 return
-            except:
+            except Exception as e:
                 logger.verbose("Incarnation is out of date. Update goalstate.")
+                msg = u"Exception updating certs: {0}".format(ustr(e))
+                detailed_msg = '{0} {1}'.format(msg, traceback.extract_tb(get_traceback(e)))
```
Comment (vrdmr) [inline azurelinuxagent/common/protocol/metadata.py]: You haven't used the `detailed_msg` or the 'msg` variable - You'll still need to add that.

```suggestion
                logger.warn(msg)
                detailed_msg = '{0} {1}'.format(msg, traceback.extract_tb(get_traceback(e)))
                logger.verbose(detailed_msg)
```

**Comment by adepue** [comment]
Cleaning up commits and will re-open when its fine

**Pull request closed without merging**